lib/impl/AzureBlobFiles.js

```diff
@@ -191,18 +209,36 @@ class AzureBlobFiles extends Files {
         const options = { delimiter: '/' }
         // prefix='' works with SAS credentials but breaks with
         // storageAccount credentials => bug in azure?
-        if (_filePath !== '') options.prefix = _filePath
+        if (_filePath !== '') {
+          options.prefix = _filePath
+        }
         const response = await this._wrapProviderRequest(azureProps.containerURL.listBlobFlatSegment(this._azure.aborter, marker, options), { filePath: _filePath })
         marker = response.marker
-        elements = elements.concat(response.segment.blobItems.map(blob => blob.name))
+        elements = elements.concat(response.segment.blobItems.map(blob => {
```

Comment:
Same here the `_listFolder's` JSDoc in `Files.js` should be updated with expected new return format (SPI).
Also `list` JSDoc return value should be updated too (API)